lib/internal/modules/package_json_reader.js

```diff
@@ -1,30 +1,101 @@
 'use strict';
 
-const { SafeMap } = primordials;
+const {
+  JSONParse,
+  ObjectPrototypeHasOwnProperty,
+  SafeMap,
+} = primordials;
+const {
+  ERR_INVALID_PACKAGE_CONFIG,
+} = require('internal/errors').codes;
 const { internalModuleReadJSON } = internalBinding('fs');
-const { pathToFileURL } = require('url');
 const { toNamespacedPath } = require('path');
+const { kEmptyObject } = require('internal/util');
+const { fileURLToPath, pathToFileURL } = require('internal/url');
 
 const cache = new SafeMap();
 
 let manifest;
 
 /**
- *
  * @param {string} jsonPath
+ * @param {{
+ *   base?: string,
+ *   specifier: string,
+ *   isESM: boolean,
+ * }} options
+ * @returns {{
+ *   exists: boolean,
+ *   pjsonPath: string,
+ *   name?: string,
+ *   main?: string,
+ *   exports?: string | Record<string, unknown>,
+ *   imports?: string | Record<string, unknown>,
+ *   type: 'commonjs' | 'module' | 'none',
+ * }}
  */
-function read(jsonPath) {
+function read(jsonPath, { base, specifier, isESM } = kEmptyObject) {
   if (cache.has(jsonPath)) {
     return cache.get(jsonPath);
   }
 
-  const { 0: string, 1: containsKeys } = internalModuleReadJSON(
+  const string = internalModuleReadJSON(
     toNamespacedPath(jsonPath),
   );
-  const result = { string, containsKeys };
-  const { getOptionValue } = require('internal/options');
+  const result = {
+    __proto__: null,
+    exists: false,
+    pjsonPath: jsonPath,
+    main: undefined,
+    name: undefined,
+    type: 'none',
```

Comment:
Assuming these are defaults to be overridden, this should be `type: 'commonjs'`.

Reply:
esm/package_config.js had the following lines:

```
  // Ignore unknown types for forwards compatibility
   if (type !== 'module' && type !== 'commonjs') {
     type = 'none';
   }
```

Reply:
Let’s port over that comment then? Because otherwise this looks like a mistake.